apps/rush-lib/src/cli/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -147,6 +162,28 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     }
 
     if (!packageDescription.dependencies.hasOwnProperty(dependencyName)) {
+      if (checkOtherProjects) {
+        // this means the current temp project doesn't provide this dependency,
+        // however, we may be able to use a different version. we prefer the latest version
+        const minimumVersion: string = '0.0.0';
+        let latestVersion: string = minimumVersion;
+
+        this.getTempProjectNames().forEach((otherTempProject: string) => {
+          const otherVersion: string | undefined = this.getDependencyVersion(dependencyName, otherTempProject, false);
+          if (otherVersion) {
+            if (semver.gt(otherVersion, latestVersion)) {
+              latestVersion = otherVersion;
+            }
+          }
+        });
+
+        if (latestVersion !== '0.0.0') {
+          // go ahead and fixup the shrinkwrap file to point at this
+          this._shrinkwrapJson.packages[tempProjectDependencyKey].dependencies[dependencyName] = latestVersion;
```

Comment:
>dependencies [](start = 66, length = 12)

can `dependencies` be undefined?

Reply:
`packages[tempProjectDependencyKey]` cannot be.

I suppose `dependencies` could be

---
In reply to: [170407066](https://github.com/Microsoft/web-build-tools/pull/543#discussion_r170407066) [](ancestors = 170407066)